runtime/lua/vim/lsp/util.lua

```diff
@@ -824,7 +825,12 @@ function M.convert_signature_help_to_markdown_lines(signature_help)
   if not signature then
     return
   end
-  vim.list_extend(contents, vim.split(signature.label, '\n', true))
+  local label = signature.label
+  if ft then
+    -- wrap inside a code block so fancy_markdown can render it properly
+    label = ("```%s\n%s\n```"):format(ft, label)
+  end
+  vim.list_extend(contents, vim.split(label, '\n', true))
```

Comment:
One disadvantage of this is it increases the size of the signature help window by 2 lines, many signature help from some language servers (`pyright` is the easiest one to test), is already only a single line, so this triples the size of the window without benefit.

More generally, I'm concerned about modifying the buffer content here thinking about GUI rendering (same concern with fancy floating markdown that I mentioned before, but now extended here since we're modifying it and bringing . Most GUIs can directly render the markdown as is.

Reply:
The `fancy_floating_window` will remove the two ` ``` ` lines, so i don't think the window height would increase? Except for the seperator?

Reply:
Ahhh that's right, it will strip it. Point 1 is fine, Point 2 is still a concern. It's fine because it does look better now, and I guess we can do a refactor of fancy floating markdown at some point to address this.

Reply:
I dont understand what you mean about changing the buffer contents in relation to the gui rendering?

This method already existed and creates the markdown for the signature. The only change here is dat we wrap the label in a code block for better rendering, similar to hover docs etc

Reply:
Also, check the `pyright` before and after example in the PR description

Reply:
I'm saying that the language server is returning valid markdown, we could transparently pass the raw markdown (without our conceal rules, inserting code fences, etc) directly to an external neovim GUI like uivonim for example, and externalize any handling of these.

Reply:
@folke sorry, that's hover though right? I was talking about signature help